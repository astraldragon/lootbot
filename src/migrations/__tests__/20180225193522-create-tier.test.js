const migration = require("../20180225193522-create-tier");
const Sequelize = require("../__mocks__/sequelize");

describe("migrations/20180225193522-create-tier", () => {
  beforeEach(() => {
    this.queryInterface = {
      createTable: jest.fn("createTable").mockReturnValue(true),
      dropTable: jest.fn("dropTable").mockReturnValue(true)
    };
  });

  it("should apply migration on up", () => {
    migration.up(this.queryInterface, Sequelize);
    expect(this.queryInterface.createTable).toMatchSnapshot();
  });

  it("should rollback migration on down", () => {
    migration.down(this.queryInterface, Sequelize);
    expect(this.queryInterface.dropTable).toMatchSnapshot();
  });
});
