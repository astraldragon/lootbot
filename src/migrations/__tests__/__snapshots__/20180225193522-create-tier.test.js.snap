// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`migrations/20180225193522-create-tier should apply migration on up 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Tiers",
      Object {
        "color": Object {
          "allowNull": false,
          "type": "integer",
        },
        "createdAt": Object {
          "allowNull": false,
          "type": "date",
        },
        "deletedAt": Object {
          "type": "date",
        },
        "guild": Object {
          "allowNull": false,
          "type": "string",
        },
        "id": Object {
          "allowNull": false,
          "autoIncrement": true,
          "primaryKey": true,
          "type": "integer",
        },
        "image": Object {
          "allowNull": false,
          "type": "string",
        },
        "luckyWeight": Object {
          "allowNull": false,
          "type": "float",
        },
        "name": Object {
          "allowNull": false,
          "type": "string",
        },
        "updatedAt": Object {
          "allowNull": false,
          "type": "date",
        },
        "weight": Object {
          "allowNull": false,
          "type": "float",
        },
      },
      Object {
        "uniqueKeys": Object {
          "tier_unique": Object {
            "customIndex": true,
            "fields": Array [
              "name",
              "guild",
              "deletedAt",
            ],
          },
        },
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;

exports[`migrations/20180225193522-create-tier should rollback migration on down 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Tiers",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": true,
    },
  ],
}
`;
